packages/firestore/src/remote/datastore.ts

```diff
@@ -16,113 +16,44 @@
  */
 
 import { CredentialsProvider } from '../api/credentials';
-import { maybeDocumentMap } from '../model/collections';
 import { MaybeDocument } from '../model/document';
 import { DocumentKey } from '../model/document_key';
 import { Mutation, MutationResult } from '../model/mutation';
 import * as api from '../protos/firestore_proto_api';
-import { hardAssert } from '../util/assert';
-import { AsyncQueue } from '../util/async_queue';
+import { debugAssert, hardAssert } from '../util/assert';
 import { Code, FirestoreError } from '../util/error';
 import { Connection } from './connection';
+import { JsonProtoSerializer } from './serializer';
 import {
-  WatchStreamListener,
-  WriteStreamListener,
   PersistentListenStream,
-  PersistentWriteStream
+  PersistentWriteStream,
+  WatchStreamListener,
+  WriteStreamListener
 } from './persistent_stream';
+import { AsyncQueue } from '../util/async_queue';
 
-import { JsonProtoSerializer } from './serializer';
-
-// The generated proto interfaces for these class are missing the database
-// field. So we add it here.
-// TODO(b/36015800): Remove this once the api generator is fixed.
-interface BatchGetDocumentsRequest extends api.BatchGetDocumentsRequest {
-  database?: string;
-}
-interface CommitRequest extends api.CommitRequest {
-  database?: string;
-}
+/**
+ * Datastore and its related methods are a wrapper around the external Google
+ * Cloud Datastore grpc API, which provides an interface that is more convenient
+ * for the rest of the client SDK architecture to consume.
+ */
+export class Datastore {}
```

Comment:
If you made this an `interface` this wouldn't even materialize in the generated code.

The downside is that empty interfaces match any object so we'd effectively lose type safety. Maybe that's why you chose class?

Reply:
Yes, an empty interface matches all objects in TypeScript. It is even commonly used to do exactly that.